### DISCUSSION TITLE
Default Types and Columns

### DISCUSSION THREAD

**Discussion by lmcardle** [supabase/supabase#3700] [Questions]
Is there a way to configure projects, such that every time a table is created specific columns, with defined types, are always created? For example, when I create a new table today, supabase automatically adds `id` and `created_at` fields. I would like to change the default type of `id` from `int8` to `uuid` with a default value of `uuid_generate_v4()` and an additional column `updated_at` with type `timestamptz`.

**Comment by soedirgo** [comment]
I don't think there's a way to configure this right now. Just curious, is your use case something like automatic table creation? Or just convenience on the UI side?

**Reply by lmcardle** [reply]
not so much automatic table creation, rather closer to the `convenience on the ui side`...however I wouldn't leave this just on the ui side. As the CLI becomes more mature and capable, I would want the same functionality there...as the UI for modeling is not where i'd want to go for production. It certainly is a `convenience` feature...however it is also a quality feature to ensure that all of my tables have x, y and z columns of types a, b and c. In my case, I want every one of my tables to always have columns `id`, `created_at` and `updated_at`. I want to avoid me or one of my devs making a mistake of omitting one of those columns or unintentionally specifying the wrong type for the column. For example, I want all `id`s to be of type UUID, not `int8`.

**Comment by FezVrasta** [comment]
Second this, to me having ids default to an integer don't make much sense and I always want to change it to UUID

**Reply by tsaxking** [reply]
I do my ids as bools

**Comment by jeydi243** [comment]
Hello, everyone. Has anyone ever suggested this as a feature?
